src/DataFrames.jl

```diff
@@ -49,6 +49,7 @@ export AbstractDataFrame,
        disallowmissing!,
        dropmissing!,
        dropmissing,
+       expandgrid,
```

Comment:
Could we find a name more closely related to `fillcombinations`, which is similar? Maybe this should even be a method of `fillcombinations`?

It would also make sense to mention the relationship between the two functions in their docstrings (dplyr does that IIRC).

Reply:
or we could use `expandcombinations` as a name if we want the name to be similar (and then export it).
However, I tend to prefer `DataFrames.product` that is documented as part of API but unexported.

Reply:
(as for the reason why I cannot call `expandgrid` in `fillcombinations` you can have a look at their codes and compare the internals of the core loop that does the expansion - there are subtle differences since there we repeat levels, but need  to allocate columns from source vector - e.g. for `CategoricalArray` it makes a difference and cannot be handled correctly if passed to `expandgrid`)